app/services/signbank/sign_search_service.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+
+module Signbank
+  class SignSearchService
+    SEARCH_SQL = <<-SQL.squish
+    WITH sign_search(id, rank_precedence, rank_order, row_num) AS
+      (SELECT rs1.id,
+              rs1.rank_precedence,
+              rs1.rank_order,
+              ROW_NUMBER() OVER (PARTITION BY rs1.id
+                                ORDER BY rs1.rank_precedence ASC) AS row_num
+      FROM
+        (SELECT words.id,
+                1 AS rank_precedence,
+                RANK() OVER (ORDER BY words.gloss) AS rank_order
+          FROM words
+          WHERE words.gloss_normalized = :term
+          UNION SELECT words.id,
+                      2 AS rank_precedence,
+                      RANK() OVER (ORDER BY words.gloss_normalized) AS rank_order
+          FROM words
+          WHERE ' ' || words.gloss_normalized || ' ' LIKE :like_term
+          UNION SELECT words.id,
+                      3 AS rank_precedence,
+                      RANK() OVER (ORDER BY words.minor_normalized) AS rank_order
+          FROM words
+          WHERE words.minor_normalized = :term
+          UNION SELECT words.id,
+                      4 AS rank_precedence,
+                      RANK() OVER (ORDER BY words.minor_normalized) AS rank_order
+          FROM words
+          WHERE ' ' || words.minor_normalized || ' ' LIKE :like_term
+          UNION SELECT words.id,
+                      5 AS rank_precedence,
+                      RANK() OVER (ORDER BY words.maori_normalized) AS rank_order
+          FROM words
+          WHERE words.maori_normalized = :term
+          UNION SELECT words.id,
+                      6 AS rank_precedence,
+                      RANK() OVER (ORDER BY words.maori_normalized) AS rank_order
+          FROM words
+          WHERE ' ' || words.maori_normalized || ' ' LIKE :like_term) AS rs1)
+    SELECT *
+    FROM sign_search
+    WHERE sign_search.row_num = 1
+    SQL
+
+    def initialize(query, relation: Sign)
+      @relation = relation
+      @query = query
+    end
+
+    def keyword_search_args
+      term = @query[:s].first
+      unaccented_term = ActiveSupport::Inflector.transliterate(term)
+      { term: unaccented_term, like_term: "% #{Sign.sanitize_sql_like(unaccented_term)} %" }
```

Comment:
Spaces are wrapped around the term to match whole words. This may need some tuning, but SQLite doesn't have a built-in regexp match function